core-tests/src/test/java/org/mule/runtime/core/internal/security/DefaultMuleSecurityManagerTestCase.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.security;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.any;
+
+
+import org.mule.runtime.api.security.Authentication;
+import org.mule.runtime.api.security.SecurityException;
+import org.mule.runtime.api.security.SecurityProviderNotFoundException;
+import org.mule.runtime.api.security.UnauthorisedException;
+import org.mule.runtime.core.api.security.SecurityManager;
+import org.mule.runtime.core.api.security.SecurityProvider;
+import org.mule.tck.junit4.AbstractMuleTestCase;
+import org.mule.tck.size.SmallTest;
+
+import java.util.Optional;
+
+import org.junit.Rule;
+import org.junit.rules.ExpectedException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+
+@SmallTest
+@RunWith(MockitoJUnitRunner.class)
+public class DefaultMuleSecurityManagerTestCase extends AbstractMuleTestCase {
+
+  @Mock
+  private SecurityProvider provider;
+
+  @Mock
+  private Authentication authentication;
+
+  @Rule
+  public ExpectedException expected = ExpectedException.none();
+
+  private SecurityManager manager;
+
+  @Before
+  public void setUp() {
+    when(provider.getName()).thenReturn("provider");
+    when(provider.supports(any())).thenReturn(true);
+    manager = new DefaultMuleSecurityManager();
+  }
+
+  @Test
+  public void authenticateCorrectly()
+          throws SecurityException, SecurityProviderNotFoundException {
+    when(provider.authenticate(authentication)).thenReturn(authentication);
+    manager.addProvider(provider);
+
+    Optional<Authentication> optional = Optional.ofNullable(manager.authenticate(authentication));
+
+    verify(provider).authenticate(authentication);
+
+    assertThat(optional.isPresent(), is(true));
+    assertThat(optional.get(), is(authentication));
+  }
+
+  @Test
+  public void authenticateWithWrongAuthentication()
+          throws SecurityException, SecurityProviderNotFoundException {
+    when(provider.authenticate(authentication)).thenThrow(UnauthorisedException.class);
+    manager.addProvider(provider);
+
+    expected.expect(UnauthorisedException.class);
+
+    manager.authenticate(authentication);
+
+    verify(provider).authenticate(authentication);
```

Comment:
... and this in a `finally`, otherwise, the exception will cause the test to end and this line will never be executed